src/main/java/com/shrralis/ssdemo1/controller/api/admin/issue/IssueManagementController.java

```diff
@@ -0,0 +1,60 @@
+package com.shrralis.ssdemo1.controller.api.admin.issue;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import com.shrralis.ssdemo1.service.interfaces.IIssueService;
+import com.shrralis.tools.model.JsonResponse;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+//@Secured({"ROLE_ADMIN"})
+@RequestMapping(value = "/admin/issues")
+public class IssueManagementController {
+
+    @Autowired
+    IIssueService issueService;
+
+    @GetMapping({"/{id}"})
+    public JsonResponse getById(@PathVariable Integer id) {
+        return new JsonResponse(issueService.findById(id));
+    }
+
+    @GetMapping({"/search/{q}"})
+    public JsonResponse getByTitleOrText(@PathVariable String q) {
+        return new JsonResponse(issueService.findByTitleOrTextContainingAllIgnoreCase(q, q));
+    }
+
+    @GetMapping({"/"})
+    public JsonResponse getIssues() {
+        return new JsonResponse(issueService.findAll());
+    }
+
+    @GetMapping({"/author/{id}"})
+    public JsonResponse getByAuthor(@PathVariable Integer id) {
+        return new JsonResponse(issueService.findByAuthor_Id(id));
```

Comment:
Underscore? Really?

Reply:
see https://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.query-methods.query-property-expressions

Reply:
O.K., so you could create `findByAuthor()` and then you could `throw` an exception if author with the id doesn't exist but if it does you can get that entity and search by that entity.
But again, as you wish.